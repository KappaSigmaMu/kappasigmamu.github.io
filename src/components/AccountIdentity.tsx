import type { ApiPromise } from '@polkadot/api'
import type { DeriveAccountRegistration } from '@polkadot/api-derive/types'
import { AccountId } from '@polkadot/types/interfaces'
import { useEffect, useState } from 'react'
import { truncateMiddle } from '../helpers/truncate'

const AccountIdentity = ({
  accountId,
  api,
  hideNotSet
}: {
  accountId: AccountId
  api: ApiPromise
  hideNotSet?: boolean
}) => {
  const [id, setId] = useState<string>('')

  useEffect(() => {
    api.derive.accounts.identity(accountId, (identity: DeriveAccountRegistration) => {
      identity.display && setId(identity.display)
    })
  }, [])

  if (!id && hideNotSet) return <></>
  if (id) return <>{id}</>
  return <>{truncateMiddle(accountId?.toString(), 20)}</>
}

export { AccountIdentity }
