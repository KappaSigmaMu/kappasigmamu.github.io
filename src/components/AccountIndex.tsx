import type { AccountIndex as PAccountIndex } from '@polkadot/types/interfaces'
import { useEffect, useState } from 'react'
import { useKusama } from '../kusama'

const AccountIndex = ({ member } : { member: SocietyMember }) => {
  const { api } = useKusama()
  const [index, setIndex] = useState<string>('')

  useEffect(() => {
    api?.derive.accounts.idToIndex(member.accountId, (accountIndex: PAccountIndex) => {
      if (accountIndex) {
        const index = api.registry.createType('AccountIndex', accountIndex.toNumber()).toString()
        setIndex(index)
      }
    })
  }, [])

  return <>{index}</>
}

export { AccountIndex }
