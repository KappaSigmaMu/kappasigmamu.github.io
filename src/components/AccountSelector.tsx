import Identicon from '@polkadot/react-identicon'
import { Dropdown, DropdownButton, Spinner } from 'react-bootstrap'
import styled from 'styled-components'
import { useAccount } from '../account/AccountContext'
import { truncateMiddle } from '../helpers/truncate'
import { useKusama } from '../kusama'

interface LevelStatusType {
  [key: string]: string
}

const LEVELSTATUS: LevelStatusType = {
  human: 'WAITING BID',
  bidder: 'BID SUBMITTED',
  candidate: 'WAITING POI',
  cyborg: ''
}

const Title = ({ activeAccount }: { activeAccount: accountType }) => {
  const { level } = useAccount()
  const account = activeAccount.name

  // TODO: Title not working
  return (
    <>
      <div className="align-top d-inline-block me-3" style={{ marginTop: 6 }}>
        <Identicon value={activeAccount.address} size={32} theme={'polkadot'} />
      </div>

      <label style={{ fontSize: '12px' }}>
        <SelectedAccountDiv className="text-start mb-1">{truncateMiddle(account || '')}</SelectedAccountDiv>
        <LevelStatusDiv>
          <label className="pe-3">JOURNEY: {level.toUpperCase()}</label>
          <label>{LEVELSTATUS[level]}</label>
        </LevelStatusDiv>
      </label>
    </>
  )
}

const Main = () => {
  const { activeAccount, setActiveAccount, accounts } = useAccount()

  const onChange = (account: string) => {
    const activeAccount = accounts.filter((acc) => acc.address.includes(account))[0]
    setActiveAccount(activeAccount)
  }

  return (
    <StyledDropdownButton
      variant="outline-light-grey"
      onSelect={(eventKey: string) => onChange(eventKey)}
      title={<Title activeAccount={activeAccount} />}
    >
      {accounts.map((option: accountType) => (
        <Dropdown.Item style={{ fontSize: '12px' }} eventKey={option.address} key={option.address} href="#">
          {option.name ? option.name : option.address}
        </Dropdown.Item>
      ))}
    </StyledDropdownButton>
  )
}

const AccountSelector = () => {
  const { api } = useKusama()
  const loading = !api?.query
  return loading ? <Spinner animation="border" variant="primary" /> : <Main />
}

const LevelStatusDiv = styled.div`
  display: flex;
  justify-content: space-between;

  label {
    color: ${(props) => props.theme.colors.grey};
    font-weight: 700;
  }
`

const SelectedAccountDiv = styled.div`
  color: ${(props) => props.theme.colors.primary};
  font-weight: 700;
  font-size: 14px;
`

const StyledDropdownButton = styled(DropdownButton)`
  background-color: ${(props) => props.theme.colors.darkGrey};
  & ::after {
    margin-bottom: 10px;
  }
`

export { AccountSelector }
