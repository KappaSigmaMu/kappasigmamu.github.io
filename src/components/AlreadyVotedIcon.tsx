import { FaCheckDouble } from 'react-icons/fa6'
import { styled } from 'styled-components'

const AlreadyVotedIcon = () => (
  <>
    <StyledCheckDouble className="me-2" />
    <label style={{ color: '#6c757d' }}>Voted</label>
  </>
)

const StyledCheckDouble = styled(FaCheckDouble)`
  flex-shrink: 0;

  & path {
    fill: ${(props) => props.theme.colors.lightGrey};
  }
`

export { AlreadyVotedIcon }
