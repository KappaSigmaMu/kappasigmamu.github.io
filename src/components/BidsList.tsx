import Identicon from '@polkadot/react-identicon'
import { Col, Row } from 'react-bootstrap'
import styled from 'styled-components'
import { humanizeBidKind, humanizeBidValue } from '../helpers/humanize'
import { truncateMiddle } from '../helpers/truncate'

const Header = () => (
  <StyledHeaderRow>
    <Col xs={1} className="text-center">#</Col>
    <Col xs={4} className="text-start">Wallet Hash</Col>
    <Col xs={5} className="text-start">Bid Kind</Col>
    <Col xs={2} className="text-end">Value</Col>
  </StyledHeaderRow>
)

const DataRow = ({ bid }: { bid: any }) => (
  <StyledDataRow key={bid.who?.toString()}>
    <Col xs={1} className="text-center">
      <Identicon value={bid.who} size={32} theme={'polkadot'} />
    </Col>
    <Col xs={4} className="text-start text-truncate">
      {truncateMiddle(bid.who?.toString())}
    </Col>
    <Col xs={5} className="text-start text-truncate">
      {humanizeBidKind(bid.kind)}
    </Col>
    <Col xs={2} className="text-end">
      {humanizeBidValue(bid.kind)}
    </Col>
  </StyledDataRow>
)

const BidsList = ({ bids }: { bids: any }): JSX.Element => (
  <>
    <Header />
    {bids.map((bid: any) => (
      <DataRow key={bid.who} bid={bid} />
    ))}
  </>
)

const StyledHeaderRow = styled(Row)`
  color: #fff;
  line-height: 3;

  & .text-end {
    padding-right: 36px;
  }
`

const StyledDataRow = styled(StyledHeaderRow)`
  background-color: #343A40;
  border-radius: 6px;
  margin-top: 10px;
`

export { BidsList }
