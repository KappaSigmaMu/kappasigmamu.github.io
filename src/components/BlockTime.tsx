import type { BlockNumber } from '@polkadot/types/interfaces'
import { formatNumber } from '@polkadot/util'
import { useBlockTime } from '../hooks/useBlockTime'
import { useKusama } from '../kusama'

const BlockTime = ({ block }: { block: BlockNumber }) => {
  const { api } = useKusama()
  const [, time] = useBlockTime(block, api)

  return (<>
    <span>{time}</span>
    &nbsp;
    <span>(#{formatNumber(block)})</span>
  </>)
}

export { BlockTime }
