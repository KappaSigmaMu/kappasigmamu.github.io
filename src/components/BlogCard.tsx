import { Card, Col } from 'react-bootstrap'

function BlogCard() {
  return (
    <Col md={3} className="mx-2">
      <Card bg={'dark'} text={'light'} style={{ width: '18rem' }} className="mb-2">
        <Card.Header>Header</Card.Header>
        <Card.Body>
          <Card.Title> Card Title </Card.Title>
          <Card.Text>
            {`Some quick example text to build on the card title and make up the
            bulk of the card's content.`}
          </Card.Text>
        </Card.Body>
      </Card>
    </Col>
  )
}

export { BlogCard }
