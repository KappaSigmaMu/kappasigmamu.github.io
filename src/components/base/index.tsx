import { Button, Row } from 'react-bootstrap'
import styled from 'styled-components'

const DataHeaderRow = styled(Row)`
  line-height: 3;

  & .text-end {
    padding-right: 36px;
  }

  * {
    color: ${(props) => props.theme.colors.lightGrey};
  }
`

const DataRow = styled(DataHeaderRow)`
  background-color: ${(props) => props.theme.colors.darkGrey};
  border-radius: 6px;
  margin-top: 10px;

  * {
  color: ${(props) => props.theme.colors.white};
}
`

const ExternalLink = styled.a.attrs(() => ({
  target: '_blank',
  rel: 'noreferrer'
}))``

const PrimaryButton = styled(Button).attrs(() => ({ variant: 'primary' }))``
const PrimaryLgButton = styled(PrimaryButton).attrs(() => ({ size: 'lg' }))``

const SecondaryLgButton = styled(Button).attrs(() => ({ variant: 'secondary', size: 'lg' }))``

export {
  DataHeaderRow,
  DataRow,
  ExternalLink,
  PrimaryButton,
  PrimaryLgButton,
  SecondaryLgButton
}
