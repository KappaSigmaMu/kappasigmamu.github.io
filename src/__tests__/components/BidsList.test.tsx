import TestRenderer from 'react-test-renderer'
import { BidsList } from '../../components/BidsList'
import { PalletSocietyBidFixture } from '../fixtures/PalletSocietyBidFixture'

const bidDeposit = PalletSocietyBidFixture.createDeposit('5GrpknVvGGrGH3EFuURXeMrWHvbpj3VfER1oX5jFtuGbfzCE', 20)
const bidVouch = PalletSocietyBidFixture.createDeposit('CxDDSH8gS7jecsxaRL9Txf8H5kqesLXAEAEgp76Yz632J9M', 10)
const bids = [bidVouch, bidDeposit]

test('renders BidsList', () => {
  const tree = TestRenderer.create(
    <BidsList bids={bids} />
  )

  expect(tree).toMatchSnapshot()
})
