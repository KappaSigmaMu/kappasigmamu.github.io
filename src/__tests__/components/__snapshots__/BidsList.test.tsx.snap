// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BidsList 1`] = `
Array [
  .c0 {
  color: #fff;
  line-height: 3;
}

.c0 .text-end {
  padding-right: 36px;
}

<div
    className="c0 row"
  >
    <div
      className="text-center col-1"
    >
      #
    </div>
    <div
      className="text-start col-4"
    >
      Wallet Hash
    </div>
    <div
      className="text-start col-5"
    >
      Bid Kind
    </div>
    <div
      className="text-end col-2"
    >
      Value
    </div>
  </div>,
  .c2 {
  cursor: copy;
  display: inline-block;
  line-height: 0;
}

.c2 > .container {
  position: relative;
}

.c2 > .container > div,
.c2 > .container > svg {
  position: relative;
}

.c2 > .container.highlight:before {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 50%;
  box-shadow: 0 0 5px 2px #aaa;
  content: '';
}

.c0 {
  color: #fff;
  line-height: 3;
}

.c0 .text-end {
  padding-right: 36px;
}

.c1 {
  background-color: #343A40;
  border-radius: 6px;
  margin-top: 10px;
}

<div
    className="c0 c1 row"
  >
    <div
      className="text-center col-1"
    >
      <div
        className="c2 ui--IdentityIcon  "
        onClick={[Function]}
      >
        <svg
          className=""
          height={32}
          id="5CSbZ7wG456oty4WoiX6a1J88VUbrCXLhrKVJ9q95BsYH4TZ"
          name="5CSbZ7wG456oty4WoiX6a1J88VUbrCXLhrKVJ9q95BsYH4TZ"
          viewBox="0 0 64 64"
          width={32}
        >
          <circle
            cx={32}
            cy={32}
            fill="#eee"
            r={32}
          />
          <circle
            cx={32}
            cy={8}
            fill="hsl(90, 94%, 35%)"
            r={5}
          />
          <circle
            cx={32}
            cy={20}
            fill="hsl(253, 94%, 15%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={14}
            fill="hsl(286, 94%, 53%)"
            r={5}
          />
          <circle
            cx={11.215390309173472}
            cy={20}
            fill="hsl(84, 94%, 15%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={26}
            fill="hsl(275, 94%, 15%)"
            r={5}
          />
          <circle
            cx={11.215390309173472}
            cy={32}
            fill="hsl(286, 94%, 53%)"
            r={5}
          />
          <circle
            cx={11.215390309173472}
            cy={44}
            fill="hsl(90, 94%, 35%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={38}
            fill="hsl(253, 94%, 15%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={50}
            fill="hsl(292, 94%, 15%)"
            r={5}
          />
          <circle
            cx={32}
            cy={56}
            fill="hsl(163, 94%, 35%)"
            r={5}
          />
          <circle
            cx={32}
            cy={44}
            fill="hsl(61, 94%, 53%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={50}
            fill="hsl(225, 94%, 35%)"
            r={5}
          />
          <circle
            cx={52.78460969082653}
            cy={44}
            fill="hsl(253, 94%, 15%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={38}
            fill="hsl(309, 94%, 75%)"
            r={5}
          />
          <circle
            cx={52.78460969082653}
            cy={32}
            fill="hsl(225, 94%, 35%)"
            r={5}
          />
          <circle
            cx={52.78460969082653}
            cy={20}
            fill="hsl(163, 94%, 35%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={26}
            fill="hsl(61, 94%, 53%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={14}
            fill="hsl(292, 94%, 15%)"
            r={5}
          />
          <circle
            cx={32}
            cy={32}
            fill="hsl(298, 94%, 15%)"
            r={5}
          />
        </svg>
      </div>
    </div>
    <div
      className="text-start text-truncate col-4"
    >
      5CSbZ7w...
    </div>
    <div
      className="text-start text-truncate col-5"
    >
      Deposit
    </div>
    <div
      className="text-end col-2"
    >
      10
    </div>
  </div>,
  .c2 {
  cursor: copy;
  display: inline-block;
  line-height: 0;
}

.c2 > .container {
  position: relative;
}

.c2 > .container > div,
.c2 > .container > svg {
  position: relative;
}

.c2 > .container.highlight:before {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  border-radius: 50%;
  box-shadow: 0 0 5px 2px #aaa;
  content: '';
}

.c0 {
  color: #fff;
  line-height: 3;
}

.c0 .text-end {
  padding-right: 36px;
}

.c1 {
  background-color: #343A40;
  border-radius: 6px;
  margin-top: 10px;
}

<div
    className="c0 c1 row"
  >
    <div
      className="text-center col-1"
    >
      <div
        className="c2 ui--IdentityIcon  "
        onClick={[Function]}
      >
        <svg
          className=""
          height={32}
          id="5GrpknVvGGrGH3EFuURXeMrWHvbpj3VfER1oX5jFtuGbfzCE"
          name="5GrpknVvGGrGH3EFuURXeMrWHvbpj3VfER1oX5jFtuGbfzCE"
          viewBox="0 0 64 64"
          width={32}
        >
          <circle
            cx={32}
            cy={32}
            fill="#eee"
            r={32}
          />
          <circle
            cx={32}
            cy={8}
            fill="hsl(118, 36%, 15%)"
            r={5}
          />
          <circle
            cx={32}
            cy={20}
            fill="hsl(258, 36%, 15%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={14}
            fill="hsl(135, 36%, 75%)"
            r={5}
          />
          <circle
            cx={11.215390309173472}
            cy={20}
            fill="hsl(354, 36%, 35%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={26}
            fill="hsl(140, 36%, 75%)"
            r={5}
          />
          <circle
            cx={11.215390309173472}
            cy={32}
            fill="hsl(258, 36%, 15%)"
            r={5}
          />
          <circle
            cx={11.215390309173472}
            cy={44}
            fill="hsl(354, 36%, 35%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={38}
            fill="hsl(140, 36%, 75%)"
            r={5}
          />
          <circle
            cx={21.607695154586736}
            cy={50}
            fill="hsl(135, 36%, 75%)"
            r={5}
          />
          <circle
            cx={32}
            cy={56}
            fill="hsl(118, 36%, 15%)"
            r={5}
          />
          <circle
            cx={32}
            cy={44}
            fill="hsl(258, 36%, 15%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={50}
            fill="hsl(22, 36%, 15%)"
            r={5}
          />
          <circle
            cx={52.78460969082653}
            cy={44}
            fill="hsl(208, 36%, 75%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={38}
            fill="hsl(286, 36%, 15%)"
            r={5}
          />
          <circle
            cx={52.78460969082653}
            cy={32}
            fill="hsl(337, 36%, 75%)"
            r={5}
          />
          <circle
            cx={52.78460969082653}
            cy={20}
            fill="hsl(208, 36%, 75%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={26}
            fill="hsl(286, 36%, 15%)"
            r={5}
          />
          <circle
            cx={42.392304845413264}
            cy={14}
            fill="hsl(22, 36%, 15%)"
            r={5}
          />
          <circle
            cx={32}
            cy={32}
            fill="hsl(16, 36%, 75%)"
            r={5}
          />
        </svg>
      </div>
    </div>
    <div
      className="text-start text-truncate col-4"
    >
      5GrpknV...
    </div>
    <div
      className="text-start text-truncate col-5"
    >
      Deposit
    </div>
    <div
      className="text-end col-2"
    >
      20
    </div>
  </div>,
]
`;
